### PR TITLE
fix int64-define-in-calc-example

### PR DESCRIPTION
**Pull request by chyroc**
if not, panic:
```go
{"1 + 0", 1}

...

	calc_test.go:28: Error in S7: INVALID(0,0), Pos(offset=4, line=1, column=5), expected one of: ( int64 
	calc_test.go:31: Error: 1 + 0 = %!d(<nil>). Expected 1
```

### CODE REVIEW

File: example/calc/calc.bnf [hunk 1]
```diff
@@ -1,15 +1,14 @@
 /* Lexical part */
 
 _digit : '0'-'9' ;
-
-int64 : '1'-'9' {_digit} ;
+int64 : '1'-'9' {_digit} | '0' ;
```
Comment (shivansh) [inline example/calc/calc.bnf]: This surely handles cases like `1 + 0` as you've noted in the testcase, but fails when dealing with say `1 + 01` (which was also the case previously).
I guess a better fix for handling all sorts of whole number notations would be -
```
int64 : '0'-'9' {_digit} ;
```
This handles both kind of testcases described above. What do you think?

**Comment by Chyroc** [comment]
@shivansh thanks for you reply, done.

**Comment by shivansh** [comment]
@Chyroc  Awesome, thanks!
Can you also please squash all commits into one, that'd make the git history cleaner.

**Comment by Chyroc** [comment]
@shivansh done, and for github, we can use `squash and merge` option:
image

**Comment by awalterschulze** [comment]
Good catch I also think `{"1 + 0", 1}` should work.

But I don't think `{"1 + 01", 2}` should work.  I think this should give an error.

**Comment by shivansh** [comment]
@awalterschulze oic ; although wouldn't `01` count as a valid whole number representation ?

**Comment by Chyroc** [comment]
but in golang  `strconv.Atoi`
```go
package main

import (
	"strconv"
	"fmt"
)

func main() {
	x, err := strconv.Atoi("01")
	if err != nil {
		panic(err)
	}

	fmt.Printf("%v\n", x)
}
```

print 1, not panic with err

**Comment by awalterschulze** [comment]
Ok fair, I think I am wrong and you are right, my bad :)

**Comment by sangisos** [comment]
Shouldn't it also support input like 0x1 in that case?

01 should be interpreted as octal right?

Try with 10, 010 and 0x10 instead to avoid false positives.

Med Vänliga Hälsningar
Alexander Andersson

**Comment by sangisos** [comment]
Atoi is only base 10 and does parse 010 to decimal 10. I expected it to work like ParseInt(s, 0, 0) for some reason. I would not like to accept leading zeros as there might be an expectation for it to be interpreted as octal.
https://golang.org/pkg/strconv/#Atoi
https://gobyexample.com/number-parsing

Sunny regards from Sweden Rock Festival

**Pull request closed without merging**